onnxruntime/python/tools/transformers/gpt2_beamsearch_tester.py

```diff
@@ -0,0 +1,603 @@
+# -------------------------------------------------------------------------
```

Comment:
Same comment: I think having it could be better to only keep `gpt2_tester.py` but (for instance, by using the factory design pattern) make it work with both `Gpt2Tester` and `Gpt2BeamSearchTester` given a use_beam_search parameter

Reply:
Same here, I have added `Gpt2TesterFactory` class. Meanwhile, I have sub-classed `Gpt2BeamSearchTester` for less code duplication